crates/drainer/src/main.rs

```diff
@@ -24,6 +26,16 @@ async fn main() -> DrainerResult<()> {
         [router_env::service_name!()],
     );
 
+    #[allow(clippy::expect_used)]
+    let web_server = Box::pin(start_web_server(conf.clone(), store.clone()))
+        .await
+        .expect("Failed to create the server");
+
+    tokio::spawn(async move {
+        let _ = web_server.await;
+        logger::error!("The health check probe stopped working!");
+    });
```

Comment:
Prefer to use [`in_current_span()`](https://docs.rs/tracing/latest/tracing/trait.Instrument.html#method.in_current_span) when using `tokio::spawn()` with a future, to propagate information about the current span. You would need the trait to be in scope, by the way.

```rust
use router_env::tracing::Instrument;
tokio::spawn(async move {
    let _ = web_server.await;
    logger::error!("The health check probe stopped working!");
}.in_current_span());
```